src/Bolt/SystemWideConnectionFactory.php

```diff
@@ -35,8 +35,17 @@ private function __construct(
     /**
      * @psalm-suppress InvalidNullableReturnType
      */
-    public static function getInstance(): SystemWideConnectionFactory
+    public static function getInstance(?string $preferredSocket = null): SystemWideConnectionFactory
```

Comment:
Let's use an enum instead of a string for better type safety and built-in documentation